### DISCUSSION TITLE
is this for coroutines only ?

### DISCUSSION THREAD

**Discussion by thomasleveil** [Krukov/cashews#128] [Q&A]
My code has both functions and coroutines, can the `@cache` decorator be used on normal functions, or is this library only works with coroutines ?

**Comment by simon-liebehenschel** [comment]
I did not use this package for a while, but as I remember, only async is supported.

You can check the [`cache` decorator code](https://github.com/Krukov/cashews/blob/master/cashews/decorators/cache/simple.py#L54) by yourself:
```python
            result = await func(*args, **kwargs)
```
The decorated function is always awaited.
